connectors/sources/sharepoint_online.py

```diff
@@ -1572,12 +1586,90 @@ async def site_lists(self, site):
 
             yield site_list
 
-    async def site_pages(self, url):
+    def _get_access_control_from_role_assignment(self, role_assignment):
+        """Extracts access control from a role assignment.
+
+        Args:
+            role_assignment (dict): dictionary representing a role assignment.
+
+        Returns:
+            access_control (list): list of usernames, which have the role assigned.
+
+        A role can be assigned to a user directly or to a group (and therefore indirectly to the users beneath).
+        If any role is assigned to a user this means at least "read" access.
+        """
+
+        def _access_control_for_user(user_):
+            user_access_control = []
+
+            user_principal_name = user_.get("UserPrincipalName")
+            login_name = _get_login_name(user_.get("LoginName"))
+
+            if user_principal_name:
+                user_access_control.append(_prefix_user(user_principal_name))
+
+            if login_name:
+                user_access_control.append(_prefix_user(login_name))
+
+            return user_access_control
```

Comment:
I'm curious about the practices of making methods method-private or module-private. Why is this one method-private particularly?

Reply:
Reasoning: I'm keeping it method-private here as I'm only using it inside the method and not outside of it. Don't know, if that's pythonic, but it makes sense to me with regards to "locality".